extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/OidcIdentityProvider.java

```diff
@@ -67,6 +67,9 @@
                 .transformToUni(new Function<TenantConfigContext, Uni<? extends SecurityIdentity>>() {
                     @Override
                     public Uni<SecurityIdentity> apply(TenantConfigContext tenantConfigContext) {
+                        // make sure we don't authenticate if this tenant is disabled
+                        if (!tenantConfigContext.oidcConfig.tenantEnabled)
+                            return Uni.createFrom().nullItem();
```

Comment:
@FroMage, Steph, this is the only change which is correct IMHO, I can help with the test a bit later on

Reply:
Hmm, how does it even reach here if `OidcAuthenticationMechanism` already does this check, I wonder if it is related to your workaround with `MpJwtValidator`...

Reply:
Don't know, but without it, it doesn't work.